src/build.ml

```diff
@@ -188,12 +199,13 @@ let get_prog = function
     >>> dyn_paths (arr (function Error _ -> [] | Ok x -> [x]))
 
 let prog_and_args ?(dir=Path.root) prog args =
-  Paths (Arg_spec.add_deps args Path.Set.empty)
+  let deps = of_deps (Arg_spec.static_deps args) in
+  dyn_path_set deps
   >>>
   (get_prog prog &&&
    (arr (Arg_spec.expand ~dir args)
     >>>
-    dyn_path_set (arr (fun (_args, deps) -> deps))
+    dyn_deps (arr (fun (_args, deps) -> deps))
```

Comment:
I realize that I just don't understand what `dyn_deps` is. I thought it was just "deps, which will be discovered when you build something". But here deps are known up-front, so what's the point of making them dynamic? (I know this is what code did before your change, which doesn't help me understand)

Reply:
Sorry, I'm dumb. I thought `arr` cancelled out the `dyn` part, but it just doesn't.